### PR TITLE
Added event for legal advisor to make bailiff decision

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1187

### Change description ###

- Adds event for legal advisor to make bailiff decision.
- Sets end state depending on if the service application type is Bailiff and if it was granted or not.
- Sets service application decision date if the application is granted.

### CODE REVIEW

**Review by mapingo — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerMakeBailiffDecision.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SYSTEMUPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.READ;
+
+@Component
+@Slf4j
+public class CaseworkerMakeBailiffDecision implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_MAKE_BAILIFF_DECISION = "caseworker-make-bailiff-decision";
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerMakeBailiffDecision.java:34]: I think it's just `caseworker-bailiff-decision` on the state diagram

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/State.java [hunk 2]
```diff
@@ -87,6 +87,13 @@
     )
     AwaitingApplicant2Response("AwaitingApplicant2Response"),
 
+    @CCD(
+        name = "Awaiting Bailiff Referral",
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/State.java:91]: Awaiting bailiff referral

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/State.java [hunk 3]
```diff
@@ -218,6 +225,14 @@
     )
     PendingRejection("PendingRejection"),
 
+
+    @CCD(
+        name = "Service Application Not Approved",
+        label = "# **${[CASE_REFERENCE]}** ${applicant1LastName} **&** ${applicant2LastName}\n### **${[STATE]}**\n",
+        access = {CaseAccessAdministrator.class}
+    )
+    ServiceApplicationNotApproved("ServiceApplicationNotApproved"),
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/State.java:234]: This one isn't required

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerMakeBailiffDecision.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+            serviceApplication.setServiceApplicationDecisionDate(LocalDate.now(clock));
+
+            if (BAILIFF.equals(serviceApplication.getServiceApplicationType())) {
+                endState = AwaitingBailiffService;
+            } else {
+                endState = AwaitingConditionalOrder;
+            }
+        } else {
+            endState = ServiceApplicationNotApproved;
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerMakeBailiffDecision.java:85]: This one should be AwaitingAos

**Review by linusnorton — APPROVED** [review]

**Pull request merged**